sdks/go/pkg/beam/runners/dataflow/dataflowlib/job.go

```diff
@@ -262,3 +272,36 @@ func printOptions(opts *JobOptions, images []string) []*displayData {
 	}
 	return ret
 }
+
+func validateWorkerSettings(opts *JobOptions) error {
+	if opts.Zone != "" && opts.WorkerRegion != "" {
+		return errors.New("cannot use option zone with workerRegion; prefer either workerZone or workerRegion")
+	}
+	if opts.Zone != "" && opts.WorkerZone != "" {
+		return errors.New("cannot use option zone with workerZone; prefer workerZone")
+	}
+	if opts.WorkerZone != "" && opts.WorkerRegion != "" {
+		return errors.New("workerRegion and workerZone options are mutually exclusive")
+	}
+
+	hasExperimentWorkerRegion := false
+	for _, experiment := range opts.Experiments {
+		if strings.HasPrefix(experiment, "worker_region") {
+			hasExperimentWorkerRegion = true
+			break
+		}
+	}
+
+	if hasExperimentWorkerRegion && opts.WorkerRegion != "" {
```

Comment:
There also needs to be a check for `hasExperimentWorkerRegion && opts.Zone != ""`.